legal-api/src/legal_api/resources/business/business.py

```diff
@@ -41,3 +45,65 @@ def get(identifier):
             return jsonify({'message': f'{identifier} not found'}), HTTPStatus.NOT_FOUND
 
         return jsonify(business=business.json())
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    def post():
+        """Create an incorporation filing, return the filing."""
+        return BusinessResource._save_incorporation_filing(request.get_json(), request)
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    def put(identifier):
+        """Edit an incorporation filing. REturn the filing."""
+        return BusinessResource._save_incorporation_filing(request.get_json(), request, identifier)
+
+    @staticmethod
+    def _save_incorporation_filing(incorporation_body, request, business_id = None):
+        """Create or update an incorporation filing. """
+        # Check that there is a JSON filing
+        if not incorporation_body:
+            return  {'message': f'No filing json data in body of post for incorporation'}, \
+                HTTPStatus.BAD_REQUEST
+        # Validate Json
+
+        temp_corp_num = incorporation_body['filing']['incorporation']['nameRequest']['nrNumber']
+        # temp_corp_num = business_id
+        # If this is an update to an incorporation filing, a temporary business identifier is passed in
+        if business_id:
+            business = Business.find_by_identifier(business_id)
+            if not business:
+                return  {'message': f'No incorporation filing exists for id {business_id}'}, \
+                HTTPStatus.BAD_REQUEST
+        else:
+            # Ensure there are no current businesses with the NR/random identifier
+            business = Business.find_by_identifier(temp_corp_num)
+            
+            if business:
+                return  {'message': f'Incorporation filing for {business_id} already exists'}, \
+                HTTPStatus.BAD_REQUEST
+            # Create an empty business record, to be updated by the filer
+            business = Business()
+            business.identifier = temp_corp_num
+            business.save()
+
+        # Ensure the business identifier matches the NR in the filing
```

Comment:
I think this should go into the validator.

Reply:
I added the soft validator to the Validator, and kept anything hitting the DB in this method. The validator will surely get more complex as more rules are discovered.